packages/dev/src/main.ts

```diff
@@ -183,6 +190,22 @@ export class NetlifyDev {
     this.#staticHandlerAdditionalDirectories = options.staticFiles?.directories ?? []
   }
 
+  private async getServerAddress(requestServerAddress?: string) {
+    if (requestServerAddress) {
+      return requestServerAddress
+    }
+
+    if (this.#server instanceof HTTPServer) {
+      return await this.#server.start()
```

Comment:
question: it's surprising that a `getServerAddress` method has the behaviour/side-effect of starting a server. is there an easy way to avoid that?

e.g. I would expect that it would be idempotent but it will throw on subsequent calls (I think)

Reply:
agreed. i wonder if HTTPServer should have a `.serving()` promise to await here, and `#server.start()` should be run somewhere else? or, if this is definitely the right place then perhaps a more dramatic verb than `get`

Reply:
We don't actually need to start a server in `getServerAddress`. I think it may have been a transient state while developing this that I forgot to clean up.

Fixed in https://github.com/netlify/primitives/pull/308/commits/ecef1bf60489ae060f44f985c0c1c179134f1efe.